src/main/java/org/parabot/randoms/ikov/QuestionSolver.java

```diff
@@ -0,0 +1,112 @@
+package org.parabot.randoms.ikov;
+
+import org.json.simple.JSONObject;
+import org.json.simple.parser.ParseException;
+import org.parabot.core.Context;
+import org.parabot.core.ui.Logger;
+import org.parabot.environment.api.utils.Time;
+import org.parabot.environment.api.utils.WebUtil;
+import org.parabot.environment.input.Keyboard;
+import org.parabot.environment.randoms.Random;
+import org.parabot.environment.randoms.RandomType;
+import org.parabot.environment.scripts.Script;
+import org.parabot.environment.scripts.framework.SleepCondition;
+import org.rev317.min.Loader;
+import org.rev317.min.api.methods.Game;
+import org.rev317.min.api.methods.Menu;
+
+import java.io.UnsupportedEncodingException;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.net.MalformedURLException;
+import java.net.URLEncoder;
+
+public class QuestionSolver implements Random {
+
+    private final int backlogID = 368, cacheID = 372;
+
+    @Override
+    public boolean activate() {
+        return Game.isLoggedIn() && Game.getOpenBackDialogId() == backlogID;
+    }
+
+    @Override
+    public void execute() {
+        String message = Loader.getClient().getInterfaceCache()[cacheID].getMessage();
+        if (!message.contains("lose items on death, beware")) {
+            message = message.replace("@dre@ ", "");
+
+            Logger.addMessage("Contacting server to get an answer", false);
+            String answer = getAnswer(message);
+
+            if (answer == null) {
+                Logger.addMessage("Could not solve the question, please report this question (and the possible answer) on the forums", true);
+                Logger.addMessage("Question: " + message, true);
+                forceRelog();
+                Logger.addMessage("Logged account out for security measures", true);
+                Context.getInstance().getRunningScript().setState(Script.STATE_STOPPED);
+            } else {
+                Logger.addMessage("Answer found, now trying: " + answer, false);
+                Menu.sendAction(679, -1, -1, 373);
+                Time.sleep(1000);
+                Keyboard.getInstance().sendKeys(answer);
+            }
+        }
+        Time.sleep(1000);
+    }
+
+    public String getAnswer(String question) {
```

Comment:
we could also fallback/override this endpoint to server config file - allowing people to add their own locally when bdn is outdated (if ever)

Reply:
Right now, if a question does not have an answer, it gets stored in the database, so we can answer it. But it might indeed be a good idea to download the list of questions&answers from the database and put it in settings folder, so users can adjust it themselves